### ISSUE TITLE
performance problem

### ISSUE THREAD

**Issue by hktalent** [blevesearch/bleve#1803]
After analysis and positioning, I found that basically I just created a few empty index libraries without any data. The code here still consumes a lot of cpu computing resources

blevesearch/bleve_index_api/analysis.go：
```
func AnalysisWorker(q AnalysisQueue) {
	// read work off the queue
	for {
		select {
		case <-q.done:
			return
		case w := <-q.queue:
			w()
		}
	}
}

```
```
goroutine 15 [select, 94 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc000be0300, 0xc000be0360})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145

goroutine 16 [select, 94 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc000be0300, 0xc000be0360})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145

goroutine 50 [select, 94 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc000be0300, 0xc000be0360})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145

goroutine 51 [select, 94 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc000be0300, 0xc000be0360})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145
```
What's wrong with this code
AI  answer：
```
There is nothing inherently wrong with this code. The function AnalysisWorker is designed to consume a queue of work items (AnalysisQueue) by executing them in a loop. It listens for a done signal and exits the loop when it receives it. If a work item is received, the function executes it.

However, there may be issues that need to be considered depending on the context in which this code is used. For example, there is no timeout mechanism built in. If the queue is empty, the function will block indefinitely, which could cause problems in a production environment. Additionally, if there are errors in the processing of work items, the function does not handle them, so they could cause problems as well.

In summary, the code itself is not necessarily wrong, but its effectiveness and stability in a production environment will depend on proper error handling and the addition of some sort of timeout mechanism.
```

**Comment by hktalent** [comment]
```
goroutine 24 [select, 91 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc00069f320, 0xc00069f380})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145

goroutine 25 [select, 91 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc00069f320, 0xc00069f380})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145

goroutine 26 [select, 91 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc00069f320, 0xc00069f380})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145

goroutine 27 [select, 91 minutes]:
github.com/blevesearch/bleve_index_api.AnalysisWorker({0xc00069f320, 0xc00069f380})
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:46 +0xbd
created by github.com/blevesearch/bleve_index_api.NewAnalysisQueue
	/Users/51pwn/MyWork/goSqlite_gorm/vendor/github.com/blevesearch/bleve_index_api/analysis.go:38 +0x145

```

**Comment by hktalent** [comment]
```
 $ go tool pprof  https://127.0.0.1:8081/debug/pprof/heap
Fetching profile over HTTP from https://127.0.0.1:8081/debug/pprof/heap
Saved profile in /Users/51pwn/pprof/pprof.alloc_objects.alloc_space.inuse_objects.inuse_space.004.pb.gz
Type: inuse_space
Time: Mar 30, 2023 at 9:15am (CST)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top
Showing nodes accounting for 1291.54MB, 94.12% of 1372.16MB total
Dropped 135 nodes (cum <= 6.86MB)
Showing top 10 nodes out of 90
      flat  flat%   sum%        cum   cum%
  739.94MB 53.93% 53.93%   739.94MB 53.93%  github.com/blevesearch/zapx/v15.(*SegmentBase).loadFieldDocValueReader
  353.65MB 25.77% 79.70%   353.65MB 25.77%  bytes.growSlice
   83.20MB  6.06% 85.76%    83.20MB  6.06%  github.com/dgraph-io/badger/skl.newArena
   27.50MB  2.00% 87.77%    32.50MB  2.37%  net.ParseCIDR
   19.80MB  1.44% 89.21%    19.80MB  1.44%  github.com/golang/snappy.Decode
   17.62MB  1.28% 90.49%    17.62MB  1.28%  go.etcd.io/bbolt.(*freelist).reindex
      15MB  1.09% 91.59%    15.51MB  1.13%  github.com/blevesearch/bleve/v2/index/scorch/mergeplan.plan
   13.62MB  0.99% 92.58%    33.43MB  2.44%  github.com/blevesearch/zapx/v15.mergeStoredAndRemap
   13.13MB  0.96% 93.54%    18.64MB  1.36%  github.com/hktalent/51pwnPlatform/pkg/server/torrentImp.Visit.func1
    8.06MB  0.59% 94.12%     8.06MB  0.59%  github.com/RoaringBitmap/roaring.newBitmapContainer
```
